homotopy-core/src/diagram.rs

```diff
@@ -291,6 +323,15 @@ impl Diagram0 {
         }
     }
 
+    #[must_use]
+    pub fn replace(&self, s: Generator, t: Generator, a: Generator) -> Self {
+        if self.generator == s || self.generator == a {
+            Self::new(t, self.orientation)
```

Comment:
I sort of imagined we would replace s and t with a (but with the dimension - 1).

Reply:
Yeah that would make sense, but it is a massive pain to implement. I think the current version is fine.